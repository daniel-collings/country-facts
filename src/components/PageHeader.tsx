interface IPageHeaderProps {
  title: string
  subtitle?: string
}

export default function PageHeader({ title, subtitle }: IPageHeaderProps) {
  return (
    <div className="p-8">
      <h1 className="text-4xl font-bold text-primaryy">{title}</h1>
      {subtitle ? (
        <p data-testid="pageHeaderSubtitle" className="text-xl text-secondary">
          {subtitle}
        </p>
      ) : null}
    </div>
  )
}
