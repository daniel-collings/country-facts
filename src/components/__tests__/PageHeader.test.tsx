import { describe, expect, test } from 'vitest'
import { render } from '@/utils/testUtils.tsx'
import PageHeader from '@/components/PageHeader.tsx'

describe('PageHeader Component', () => {
  test('should render correctly', () => {
    const { asFragment } = render(<PageHeader title="Hello World" />)
    expect(asFragment()).toMatchSnapshot()
  })
})
