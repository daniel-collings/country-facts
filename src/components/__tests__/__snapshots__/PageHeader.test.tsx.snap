// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PageHeader Component > should render component correctly 1`] = `
<DocumentFragment>
  <div
    class="h-full min-h-screen bg-base-100 p-4 sm:p-8"
    data-theme="light"
  >
    <main
      class="max-w-7xl mx-auto px-0 sm:px-6 lg:px-8"
    >
      <div
        class="p-8"
      >
        <h1
          class="text-4xl font-bold text-primaryy"
        >
          Hello World
        </h1>
      </div>
    </main>
  </div>
</DocumentFragment>
`;
